### ISSUE TITLE
Hidden entities are not hidden

### ISSUE THREAD

**Issue by lindhe** [home-assistant/frontend#22040]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

I have some devices with switch entities that I want to "show as Light" rather than as a Switch. When I do that, the Switch entity is marked as "hidden" in favor of the new virtual Light entity:

![Screenshot showing the Switch configuration](https://github.com/user-attachments/assets/31e903f6-cd34-409e-a38c-34bad12cc10e)
![Screenshot of the device](https://github.com/user-attachments/assets/305f04f3-937a-4caf-b3d8-beda2a657486)

Yet, both devices are visible when interacting with the device:

![Screenshot showing the dropdown menu in a Script](https://github.com/user-attachments/assets/effc9a63-18d4-441a-9326-22a2bf90a364)

Notice how both "Light" and "Switch" devices get listed. This clutters the UI and makes it harder to navigate.

### Describe the behavior you expected

I expect that these Switch entities to not show up in Home Assistant by default in almost any view. The only exception, I think, should be when I go to the "Entities" list and disable the filter that filters out hidden devices.

### Steps to reproduce the issue

1. Have a Device with a Switch entity (or similar).
2. [Edit the entity](https://www.home-assistant.io/docs/configuration/customizing-devices/) and select "Show as Light" (or similar).
3. Click update.
4. Notice how the Switch entity is marked as "Hidden" and the Light entity is not.
5. Go to a Script or Automation and do something with the Device.
6. Scroll in the list of available actions and notice how both entities are shown in the list.

### What version of Home Assistant Core has the issue?

core-2024.9.2

### In which browser are you experiencing the issue with?

Firefox 129.0.2

### Which operating system are you using to run this browser?

Ubuntu 22.04

**Comment by karwosts** [comment]
Hidden I think only is intended to remove entities from autogenerated dashboard, it's not to hide it from all pickers. 

Maybe that could be changed but I don't think that's the intention.

**Comment by lindhe** [comment]
You are probably right. To me, this is confusing and I think it would improve the user experience if we changed that. Scrolling that list of entities is really an edge case, compared to all the configuration one does for scripts, automations and dashboards (yes, I found it works the same in the dropdown for dashboards too).

**Comment by matthiasdebaat** [comment]
I agree that the labeling is confusing. The description tries to make it better, but from research we know that people don't read. A quick solution for now could be to change the label to `Visible on default dashboard`. 

We're in the process of making dashboarding better. We're going to include this pain point.

**Comment by karwosts** [comment]
I know area card is one more place where this is used, so it's not _only_ for visible on default dashboard.

**Comment by lindhe** [comment]
> from research we know that people don't read.

So true! 😄 

Sorry for me being ignorant here, but what's the "default dashboard"? Do you mean the autogenerated one? If that is what you mean: great, you've found another irregularity!

Then we have three places where "hide" does work:

- The default dashboard.
- The area card.
- The ["Entites" table view](https://my.home-assistant.io/redirect/entities/).

Maybe the places where I have found that it does not work are all instances of the same functionality: picking entities from a drop-down list. If so, it would be interesting to look into the code that handles indexing of that list (I assume it's cached and not calculated each time a drop-down menu is shown).

Does anyone know where to start looking?

**Comment by lindhe** [comment]
It would be nice to know if the maintainers deem this to be a reasonable request or if it's not relevant to change this design.

**Comment by matthiasdebaat** [comment]
> It would be nice to know if the maintainers deem this to be a reasonable request or if it's not relevant to change this design.

https://github.com/home-assistant/frontend/issues/22040#issuecomment-2367588004
